opwen_email_client/webapp/forms/register.py

```diff
@@ -0,0 +1,52 @@
+from os import getenv
+from json import dumps
+from pathlib import Path
+from shutil import chown
+
+from flask_wtf import FlaskForm
+from requests import post
+from wtforms import StringField
+from wtforms import SubmitField
+
+from opwen_email_client.webapp.tasks import register
+
+
+class RegisterForm(FlaskForm):
+
+    client_name = StringField()
+    github_username = StringField()
+    github_token = StringField()
+    submit = SubmitField()
+
+    def register_client(self):
+        path = self._setup_path()
+        self._setup_client(path)
+
+    def _setup_client(self, path):
+        name = self.client_name.data.strip()
+        token = self.github_token.data.strip()
+
+        client_domain = '{}.{}'.format(name, 'lokole.ca')
+        client_create_url = 'https://{}/api/email/register/'.format('mailserver.lokole.ca')
+        create_request_payload = dumps({'domain': client_domain}).encode('utf-8')
+        create_headers = {'Content-Type': 'application/json',
+                          'Content-Length': str(len(create_request_payload)),
+                          'Authorization': 'Bearer {}'.format(token)}
+
+        response = post(client_create_url, data=create_request_payload, headers=create_headers)
+        response.raise_for_status()
```

Comment:
How does error handling work if the request fails? Does the form have to catch the response error and wrap it in a [ValidationError](https://wtforms.readthedocs.io/en/2.3.x/validators/?highlight=error#wtforms.validators.ValidationError) for the problem to be displayed to the user?

Reply:
I was actually going to ask you about how to display the error to the user. The linked page is what I have been searching for. Wrapping it in a ValidationError is the best bet I think!